test/device/cuda.jl

```diff
@@ -981,6 +981,18 @@ end
     end
 end
 
+@testset "exceptions" begin
+    @testset for T in [Int32, Int64, UInt32, UInt64]
+        a = CuArray([typemax(T)])
+        function kernal(T, a)
+            @atomic a[1] = 1
+            @atomic a[1] = "1"
+            @atomic a[1] = i
+            return
+        end
+        @cuda threads=32 kernal(T, a)
+        @test Array(a)[1] == 1
+    end
```

Comment:
Did you run these tests? They are intended to fail, so you can't just run them (the test would fail). Instead, you need to `test_throws`. Also, you can't put multiple failing atomic expressions in a single test because it'll bail out at the first error.
I'll push a fix, since it's somewhat finicky to use `test_throws` with macros.

Reply:
I did test them, but they didn't throw any error